server/index.js

```diff
@@ -29,6 +29,27 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const { productId } = req.params;
+  const sql = `
+    select *
+    from "products"
+    where "productId" = $1
+  `;
+
+  const params = [productId];
+  if (productId.length < productId || productId <= 0) {
+    return next(new ClientError('Invalid product', 400));
+  } else {
+    res.status(500).json({
+      error: 'an unexpected error occurred'
+    });
+  }
+  db.query(sql, params)
+    .then(result => res.json(result.rows[0]))
+    .catch(err => next(err));
```

Comment:
If no product exists with that `productId`, the client should receive a `404` response.